src/ray/common/client_connection.h

```diff
@@ -125,6 +125,11 @@ class ServerConnection : public std::enable_shared_from_this<ServerConnection> {
 
   std::string DebugString() const;
 
+  void AsyncWaitTerminated(std::function<void()> callback) {
+    socket_.async_wait(local_stream_socket::wait_type::wait_error,
```

Comment:
nit: add comment? Is this waiting for the socket to be cleaned up (`TIME_WAIT`)?